File: docs/src/regex.rb
#!/usr/bin/env ruby

puts('Test Ruby regex')
puts("Must be false: ", !! /^wrong$/.match("hello"))
puts("Must be true: ", !! /^hello$/.match("hello"))
puts("True if permissive: ", !! /^hello$/.match("hello\n"))
puts("Should be true ($ always multi): ", !! /^hello$/.match("hello\nthere"))

File: docs/src/regex.py
#!/usr/bin/env python3

import re

print('Test Python regex')
print("Must be false: ", bool(re.search(r'^wrong$', "hello")))
print("Must be true: ", bool(re.search(r'^hello$', "hello")))
print("True if permissive: ", bool(re.search(r'^hello$', "hello\n")))
print("Should be false: ", bool(re.search(r'^hello$', "hello\nthere")))
